packages/liveblocks-react/src/room.tsx

```diff
@@ -2356,6 +2310,52 @@ export function createRoomContext<
   });
 }
 
+const DEFAULT_DEDUPING_INTERVAL = 2000; // 2 seconds
+
+const lastRequestedAtByRoom = new Map<string, Date>(); // A map of room ids to the timestamp when the last request for threads updates was made
+const requestStatusByRoom = new Map<string, boolean>(); // A map of room ids to a boolean indicating whether a request to retrieve threads updates is in progress
```

Comment:
There are no code changes in this entire block, I just moved it out of the factory body and made `client` a parameter.

The question here now is: is this a safe operation? Especially: will this be safe to do regarding the `lastRequestedAtByRoom` and `requestStatusByRoom` caches? Since those are no longer scoped to the factory instance, but now shared globally.

Will it be important to create a new cache for every `client` instance?